clients/callback_client.go

```diff
@@ -157,3 +176,52 @@ func (ts TranscodeStatus) String() string {
 	}
 	return "unknown"
 }
+
+// The various status messages we can send
+
+type TranscodeStatusMessage struct {
+	CompletionRatio float64 `json:"completion_ratio"` // No omitempty or we lose this for 0% completion case
+	Error           string  `json:"error,omitempty"`
+	Retriable       *bool   `json:"retriable,omitempty"` // Has to be a pointer or we can't differentiate omission from 'false'
+	Status          string  `json:"status,omitempty"`
+	Timestamp       int64   `json:"timestamp"`
+}
+
+type InputVideoTrack struct {
```

Comment:
Missing `start_time` here, isn't it relevant? Maybe not if we handle it correctly I guess (which we don't do in current pipeline 😆) but raising it here for consideration.

Reply:
Sure, what unit would it be?

Reply:
I think it's in floating seconds like the duration

Reply:
Added